Reading with sox and offset returns wrong duration

If you read a file format, that is not WAV, FLAC, OGG, but supported by `sox` and you ask for `offset` without specifying `duration`, the duration of the returned file can be wrong.

In the tests an MP3 file is created with one or two channels and 48000 Hz sampling rate. When reading this using `sox` the returned duration is wrong only for the two channel case:
```
        sig, fs = af.read(mp3_file, offset=offset)
        assert_allclose(
            _duration(sig, sampling_rate),
            af.duration(mp3_file) - offset,
            rtol=0,              
>           atol=tolerance('duration', sampling_rate),
        )
E       AssertionError:
E       Not equal to tolerance rtol=0, atol=2.08333e-05
E
E       Mismatched elements: 1 / 1 (100%)  
E       Max absolute difference: 0.024      
E       Max relative difference: 0.02643172
E        x: array(0.884)
E        y: array(0.908)
```

--- a/tests/test_audiofile.py
+++ b/tests/test_audiofile.py
@@ -447,15 +447,12 @@ def test_mp3(tmpdir, magnitude, sampling_rate, channels):
     sig, fs = af.read(mp3_file, offset=offset, duration=duration)
     assert _duration(sig, sampling_rate) == duration
     sig, fs = af.read(mp3_file, offset=offset)
-    # Don't test for 48000 Hz and 2 channels
-    # https://github.com/audeering/audiofile/issues/23
-    if not (sampling_rate == 48000 and channels == 2):
-        assert_allclose(
-            _duration(sig, sampling_rate),
-            af.duration(mp3_file) - offset,
-            rtol=0,
-            atol=tolerance('duration', sampling_rate),
-        )
+    assert_allclose(
+        _duration(sig, sampling_rate),
+        af.duration(mp3_file) - offset,
+        rtol=0,
+        atol=tolerance('duration', sampling_rate),
+    )
 
 
 def test_formats():

